auction/contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "./AuctionHouseBase.sol";
+
+import "@rarible/exchange-v2/contracts/lib/LibTransfer.sol";
+import "@rarible/exchange-v2/contracts/RaribleTransferManager.sol";
+
+/// @dev contract to create and interact with auctions 
+contract AuctionHouse is AuctionHouseBase, TransferExecutor,  RaribleTransferManager{
+    using LibTransfer for address;
+
+    /// @dev mapping to store data of auctions for auctionId
+    mapping(uint => Auction) auctions;
+
+    /// @dev latest auctionId
+    uint256 private auctionId;          //unic. auction id
+
+    /// @dev minimal auction duration and also the time for that auction is extended when it's about to end (endTime - now < EXTENSION_DURATION)
+    uint256 private constant EXTENSION_DURATION = 15 minutes;
+
+    /// @dev maximum auction duration
+    uint256 private constant MAX_DURATION = 1000 days;
+
+    function __AuctionHouse_init(
+        INftTransferProxy _transferProxy,
+        IERC20TransferProxy _erc20TransferProxy,
+        uint newProtocolFee,
+        address newDefaultFeeReceiver,
+        IRoyaltiesProvider newRoyaltiesProvider
+    ) external initializer {
+        __Context_init_unchained();
+        __Ownable_init_unchained();
+        __TransferExecutor_init_unchained(_transferProxy, _erc20TransferProxy);
+        __RaribleTransferManager_init_unchained(newProtocolFee, newDefaultFeeReceiver, newRoyaltiesProvider);
+        __AuctionHouseBase_init();
+        __AuctionHouse_init_unchained();
+    }
+
+     function __AuctionHouse_init_unchained() internal initializer {
+        auctionId = 1;
+    }
+
+    /// @dev creates an auction and locks sell asset
+    function startAuction(
+        LibAsset.Asset memory _sellAsset,
+        LibAsset.AssetType memory _buyAsset,
+        uint minimalStep,
+        uint minimalPrice,
+        bytes4 dataType,
+        bytes memory data
+    ) external {
+        uint currentAuctionId = getNextAndIncrementAuctionId();
```

Comment:
if we move the initialization of currentAuctionId to right before its usage, we will increment auction id only if all requirements are passed

Reply:
fixed